nsim-nsim_vpx5: libraries.libc tests build failures

### Describe the bug

Multiple tests are failing on build failure on nsim-nsim_vpx5:
- libraries.libc.c11_threads.minimal
- libraries.libc.c11_threads.picolibc.module
- libraries.libc.c11_threads.picolibc.notls
- libraries.libc.sprintf

Error is the same on all them: "Build failure - error: typedef redefinition with different types ('unsigned long' vs 'int')"

### Regression

- [x] This is a regression.

### Steps to reproduce

west twister -p nsim/nsim_vpx5 -s libraries.libc.sprintf
or
west build -p -b nsim/nsim_vpx5 tests/lib/sprintf -T libraries.libc.sprintf

### Relevant log output

```shell
INFO    -  83/295 nsim/nsim_vpx5            libraries.libc.c11_threads.minimal                 ERROR Build failure (nsim <arcmwdt>)
INFO    - /.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/build.log
ERROR   - Loading Zephyr module(s) (Zephyr base): zephyr_default:kconfig
-- Application: /.../tests/lib/c_lib/thrd
-- CMake version: 3.28.3
-- Found Python3: /opt/python/venv/bin/python3 (found suitable version "3.12.3", minimum required is "3.12") found components: Interpreter 
-- Cache files will be written to: /.../.cache
-- Zephyr version: 4.3.99
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: nsim, qualifiers: nsim_vpx5
-- Found host-tools: zephyr 1.0.0 (/opt/toolchains/zephyr-sdk-1.0.0)
-- Found host-tools: zephyr 1.0.0 (/opt/toolchains/zephyr-sdk-1.0.0)
-- Found dts preprocessor: /opt/toolchains/zephyr-sdk-1.0.0/gnu/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc (Zephyr SDK 1.0.0)
-- Found Dtc: /opt/toolchains/zephyr-sdk-1.0.0/hosttools/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /.../boards/snps/nsim/arc_classic/nsim_nsim_vpx5.dts
-- Generated zephyr.dts: /.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/zephyr.dts
-- Generated pickled edt: /.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/edt.pickle
-- Generated devicetree_generated.h: /.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/include/generated/zephyr/devicetree_generated.h

warning: MAX_THREAD_BYTES (defined at arch/Kconfig:410) was assigned the value '8' but got the value
''. Check these unsatisfied dependencies: USERSPACE (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MAX_THREAD_BYTES and/or look up
MAX_THREAD_BYTES in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.


warning: MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS (defined at lib/libc/minimal/Kconfig:6) was assigned
the value 'y' but got the value 'n'. Check these unsatisfied dependencies: MINIMAL_LIBC (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS and/or
look up MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.


warning: MINIMAL_LIBC_RAND (defined at lib/libc/minimal/Kconfig:30) was assigned the value 'y' but
got the value 'n'. Check these unsatisfied dependencies: MINIMAL_LIBC (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MINIMAL_LIBC_RAND and/or look up
MINIMAL_LIBC_RAND in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.


warning: The choice symbol MINIMAL_LIBC (defined at lib/libc/Kconfig:86) was selected (set =y), but
ARCMWDT_LIBC (defined at lib/libc/Kconfig:135) ended up as the choice selection. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MINIMAL_LIBC and/or look up MINIMAL_LIBC in
the menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration
Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful too.

...

[13/192] Building C object CMakeFiles/app.dir/src/cnd.c.obj
FAILED: CMakeFiles/app.dir/src/cnd.c.obj 
/.../MetaWare_MX_1_2/MetaWare/arc/bin/ccac -DKERNEL -DK_HEAP_MEM_POOL_SIZE=512 -DSOC_MMU_PAGE_SIZE=0x10000 -DTC_RUNID=a7b54cba86280e5c32d42715681d7f05 -DZVFS_OPEN_SIZE=4 -D_POSIX_CLOCK_SELECTION=200809L -D_POSIX_CPUTIME=200809L -D_POSIX_C_SOURCE=200809L -D_POSIX_MONOTONIC_CLOCK=200809L -D_POSIX_MULTI_PROCESS=200809L -D_POSIX_REALTIME_SIGNALS=200809L -D_POSIX_THREADS=200809L -D_POSIX_THREAD_SAFE_FUNCTIONS=200809L -D_POSIX_TIMERS=200809L -D__DECLARE_RCC_ATOMIC_ENV -D__DECLARE_RCC_RC_ATOMIC_ENV -D__ZEPHYR__=1 -I/.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/include/generated/zephyr -I/.../tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/include/generated -I/.../soc/snps/nsim/arc_classic -I/.../lib/posix/c_lib_ext/getopt -I/.../include/zephyr/posix -I/.../subsys/testsuite/include -I/.../subsys/testsuite/coverage -I/.../subsys/testsuite/ztest/include -I/.../drivers -I/__w/os.rtos.zephyr.zephyr-intel/intel/zephyr-intel/include -isystem /.../lib/libc/arcmwdt/include -isystem /.../lib/libc/common/include -fno-strict-aliasing -Werror -Os -imacros /.../twister-out/nsim_nsim_vpx5/arcmwdt/tests/lib/c_lib/thrd/libraries.libc.c11_threads.minimal/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -Hnocopyr -HL -Hnosdata -Hoff=Stackcheck_alloca -imacros /.../include/zephyr/toolchain/zephyr_stdint.h -Wformat -Wformat-security -Wno-format-zero-length -Wno-unaligned-pointer-conversion -Wno-incompatible-pointer-types-discards-qualifiers -Wno-typedef-redefinition -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wno-sometimes-uninitialized -Wno-shift-overflow -Wno-missing-braces -Wno-self-assign -Wno-address-of-packed-member -Wno-unused-function -Wno-initializer-overrides -Wno-section -Wno-unknown-warning-option -Wno-unused-variable -Wno-format-invalid-specifier -Wno-gnu -Wno-tautological-compare -Werror=implicit-int -Wno-vla -fmacro-prefix-map=/.../tests/lib/c_lib/thrd=CMAKE_SOURCE_DIR -fmacro-prefix-map=ZEPHYR_BASE -fmacro-prefix-map=WEST_TOPDIR -ffunction-sections -fdata-sections -gdwarf-2 -arcv2hs -core4 -uarch_rev=1:4 -Xcode_density -Xatomic -Xll64 -Xunaligned -Xdiv_rem=radix4 -Xswap -Xbitscan -Xmpy_option=qmpyh -Xshift_assist -Xbarrel_shifter -Xtimer0 -Xtimer1 -Xrtc -dcache=32768,64,2,a -Hld_cycles=1 -DDCCM_SYSTEM_BASE_CORE0=0x80000000 -Hccm -DICCM0_SYSTEM_BASE_CORE0=0x0000000 -Xstu=4 -Xvdsp4 -Xvec_unit_rev_minor=1 -Xvec_width=512 -Xvec_mem_size=256k -Xvec_mem_bank_width=16 -Xvec_max_fetch_size=16 -Xvec_num_slots=3 -Xvec_super_with_scalar -Xvec_regs=40 -Xvec_num_rd_ports=6 -Xvec_num_acc=8 -Xvec_num_mpy=2 -Xvec_mpy32 -Xvec_num_alu=3 -Xvec_guard_bit_option=2 -Xvec_stack_check -DVEC_MEM_SYS_BASE_CORE0=0xb4000000 -std=c17 -MD -MT CMakeFiles/app.dir/src/cnd.c.obj -MF CMakeFiles/app.dir/src/cnd.c.obj.d -o CMakeFiles/app.dir/src/cnd.c.obj -c /.../tests/lib/c_lib/thrd/src/cnd.c
In file included from /.../tests/lib/c_lib/thrd/src/cnd.c:7:
In file included from /.../tests/lib/c_lib/thrd/src/thrd.h:11:
In file included from /.../lib/libc/arcmwdt/include/time.h:12:
/.../include/zephyr/posix/posix_time.h:26:23: error: typedef redefinition with different types ('unsigned long' vs 'int')
typedef unsigned long clockid_t;
                      ^
/.../MetaWare_MX_1_2/MetaWare/arc/inc/time.h:214:21: note: previous definition is here
        typedef int clockid_t;
                    ^
1 error generated.
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Zephyr version: v4.3.0-9315-ge08a582369da
Toolchain: zephyr-sdk-1.0.0

### Additional Context

_No response_

--- a/lib/libc/arcmwdt/include/signal.h
+++ b/lib/libc/arcmwdt/include/signal.h
@@ -8,6 +8,8 @@
 #ifndef LIB_LIBC_ARCMWDT_INCLUDE_SIGNAL_H_
 #define LIB_LIBC_ARCMWDT_INCLUDE_SIGNAL_H_
 
+#include <sys/types.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif

--- a/lib/libc/arcmwdt/include/string.h
+++ b/lib/libc/arcmwdt/include/string.h
@@ -14,6 +14,7 @@ extern "C" {
 #endif
 
 extern size_t strnlen(const char *s, size_t maxlen);
+extern char *strsignal(int signum);
 
 #ifdef __cplusplus
 }

--- a/lib/libc/arcmwdt/include/sys/types.h
+++ b/lib/libc/arcmwdt/include/sys/types.h
@@ -17,6 +17,7 @@
 #define _NLINK_T_DECLARED
 #define _UID_T_DECLARED
 #define _GID_T_DECLARED
+#define _PID_T_DECLARED
 
 #ifndef _SSIZE_T_DEFINED
 #define _SSIZE_T_DEFINED

--- a/lib/libc/arcmwdt/include/time.h
+++ b/lib/libc/arcmwdt/include/time.h
@@ -8,6 +8,7 @@
 #define LIB_LIBC_ARCMWDT_INCLUDE_TIME_H_
 
 #include_next <time.h>
+#include <sys/types.h>
 
 #include <zephyr/posix/posix_time.h>
 

